### DISCUSSION TITLE
Support for arbitrarily large integers (ie. Matrix of BigInteger)

### DISCUSSION THREAD

**Discussion by Superbest** [mathnet/mathnet-numerics#804] [Ideas]
Is support for a matrix made up of BigInteger planned?

If it is not planned, it should be. If it already exists, it needs to be documented more explicitly.

**Comment by MaLiN2223** [comment]
@cdrnet How about creating matrix for generic type? I did it [here](https://github.com/MaLiN2223/Malins/blob/master/C%23/DataStructures/Matrices/Generics/Matrix.cs) and I think it might be usefull because creating class Matrix : Matrix <type> for every type might be exhausting.

**Comment by MovGP0** [comment]
there should be support for arbitrary large floating-point-decimals too. we could use [the W3b.Sine library](https://github.com/vogon/Demotic/tree/master/W3b.Sine) which provides the elementary math functions. 

Unfortunately, the library was never completed, so there might be the need for some additional work.  Ie. it might be needet to calculate π with arbitrary precision (using the [Chudnovsky algorithm](https://en.wikipedia.org/wiki/Chudnovsky_algorithm)) or other mathematical constants. 

An alternative library would be "GNU Multiple Precision Floating-Point Reliably" ([GNU MPFR .NET Wrapper](https://github.com/emphasis87/mpfr.NET)).

**Comment by MovGP0** [comment]
However, since those libraries require to give the precision ahead of time of the calculations, an re-implementation might be needed, such that additional digits can get requested as needed using `IEnumerator.MoveNext()`.
